src/datasources/prices-api/coingecko-api.service.ts

```diff
@@ -132,6 +141,15 @@ export class CoingeckoApi implements IPricesApi {
     }
   }
 
+  async registerNotFoundTokenPrice(args: {
+    chainName: string;
+    tokenAddress: string;
+    fiatCode: string;
+  }): Promise<void> {
+    const cacheDir = CacheRouter.getTokenPriceCacheDir(args);
+    return this.cacheService.expire(cacheDir.key, this.notFoundPriceTtlSeconds);
+  }
```

Comment:
What's the status code returned when a token is not found? The `CacheFirstDataSource` has an expiration time that for those cases.

Reply:
The Coingecko API always returns a 200 status code even if the token is not found. That makes it harder to use the API for this use case of requesting the token prices one by one. It simply returns an empty object as payload in those cases. That's why checking the response actually contains the price is done in the repository.

My best guess around this design decision on the Coingecko side was taken because the endpoint expects several addresses to be sent in each call (as I'm implementing on another PR). So some addresses may be found and others may not, and the response is always a 200 containing an object holding a subset of the addresses requested.

Reply:
OK makes sense. Thanks for explaining. Since this should not be considered a final solution I would say that it's fine and we can re-work the domain/datasource API on a second iteration!